### ISSUE TITLE
Chrome users are logged out of code.org when the browser is closed

### ISSUE THREAD

**Issue by snickell** [code-dot-org/code-dot-org#56774]
Users of other browsers stay logged into code.org (Edge, Firefox), but it seems Chrome users are logged out whenever the browser closes.

Per https://developer.chrome.com/blog/cookie-max-age-expires :
> As of Chrome release [M104 (August 2022)](https://chromiumdash.appspot.com/schedule) cookies can no longer set an expiration date more than 400 days in the future.
>
> This change does not impact session cookies—cookies that do not explicitly set an expiration date with Max-Age or Expires—as these are instead cleared when the browsing session ends.

This seems potentially related.

I'm not sure if we overload the session cookie for auth as well (???), but at the very least, we set no expiration on it, so if it is related to auth chrome would be clearing it on each browser close: https://github.com/code-dot-org/code-dot-org/blob/fdf7a2efa1e634497e993f6fae48e216dbe99c58/dashboard/config/initializers/session_store.rb/#L3-L6

**Comment by snickell** [comment]
We're using devise for auth, and its entirely possible it sets its own cookies, which could ALSO be the source of the surprise expiry, see: https://stackoverflow.com/questions/26564174/user-logged-out-automatically-when-browser-is-closed-when-using-devise-gem

1. We do enable devise's :rememberable hooks in user.rb: https://github.com/code-dot-org/code-dot-org/blob/fdf7a2efa1e634497e993f6fae48e216dbe99c58/dashboard/app/models/user.rb/#L174-L175
2. In `devise.rb` we configure the browser to be remembered for 2 weeks (https://github.com/code-dot-org/code-dot-org/blob/fdf7a2efa1e634497e993f6fae48e216dbe99c58/dashboard/config/initializers/devise.rb/#L170-L172):
```
  # ==> Configuration for :rememberable
  # The time the user will be remembered without asking for credentials again.
  config.remember_for = 2.weeks
```

When logged out of studio.code.org, here's the potentially relevant cookies:
<img width="483" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/223277/06a1bb21-9e73-4d5b-a3b8-9172c666bcf1">

**Comment by snickell** [comment]
Changing the session store cookie to a 200 day expiration ("one login per school year") results in being able to quit chrome, and when its re-opened we're still logged in:
```
Dashboard::Application.config.session_store :cookie_store,
  key: session_cookie_key,
  secure: !CDO.no_https_store && (!Rails.env.development? || CDO.https_development),
  domain: :all,
  expire_after: 200.days # don't expire in the same school year
```

It appears that devise is NOT setting its own cookies, instead it appears to be re-using the session store.

**Comment by snickell** [comment]
PR: https://github.com/code-dot-org/code-dot-org/pull/56777

**Comment by snickell** [comment]
Fixed by PR #56777
